docs/manual/scala/guide/services/code/ServiceImplementation.scala

```diff
@@ -0,0 +1,204 @@
+
+    override def tick(intervalMs: Int) = ServiceCall { tickMessage =>
+      Future.successful(Source.tick(
+        intervalMs.milliseconds, intervalMs.milliseconds, tickMessage
+      ))
+    }
+    //#tick-service-call
+  }
+}
+
+package hellostream {
+
+  import com.lightbend.lagom.scaladsl.api.ServiceCall
+  import scala.concurrent.Future
+
+  class HelloStreamImpl extends HelloStream {
+    //#hello-service-call
+    override def sayHello = ServiceCall { names =>
+      Future.successful(names.map(name => s"Hello $name!"))
+    }
+    //#hello-service-call
+  }
+}
+
+package serverservicecall {
+
+  import scala.concurrent.Future
+
+  class UsingServerServiceCall extends helloservice.HelloService {
+    //#server-service-call
+    import com.lightbend.lagom.scaladsl.server.ServerServiceCall
+    import com.lightbend.lagom.scaladsl.api.transport.ResponseHeader
+
+    override def sayHello = ServerServiceCall { (requestHeader, name) =>
+      val user = requestHeader.principal
+        .map(_.getName).getOrElse("No one")
+      val response = s"$user wants to say hello to $name"
+
+      val responseHeader = ResponseHeader.Ok
+        .withHeader("Server", "Hello service")
+
+      Future.successful((responseHeader, response))
+    }
+    //#server-service-call
+  }
+}
+
+package servicecallcomposition {
+
+  import scala.concurrent.{ExecutionContext, Future}
+  import com.lightbend.lagom.scaladsl.server.ServerServiceCall
+
+  object Logging {
+    //#logging-service-call
+    def logged[Request, Response](serviceCall: ServerServiceCall[Request, Response]) =
+      ServerServiceCall.compose { requestHeader =>
+        println(s"Received ${requestHeader.method} ${requestHeader.uri}")
+        serviceCall
+      }
+    //#logging-service-call
+  }
+
+  class LoggedHelloService extends helloservice.HelloService {
+    import Logging.logged
+
+    //#logged-hello-service
+    override def sayHello = logged(ServerServiceCall { name =>
+      Future.successful(s"Hello $name!")
+    })
+    //#logged-hello-service
+  }
+
+  trait User
+
+  //#user-storage
+  trait UserStorage {
+    def lookupUser(username: String): Future[Option[User]]
+  }
+  //#user-storage
+
+  object Authentication {
+    val userStorage: UserStorage = _
+    implicit val ec: ExecutionContext = _
+
+    //#auth-service-call
+    import com.lightbend.lagom.scaladsl.api.transport.Forbidden
+
+    def authenticated[Request, Response](
+      serviceCall: User => ServerServiceCall[Request, Response]
+    ) = ServerServiceCall.composeAsync { requestHeader =>
+
+      // First lookup user
+      val userLookup = requestHeader.principal
+        .map(principal => userStorage.lookupUser(principal.getName))
+        .getOrElse(Future.successful(None))
+
+      // Then, if it exists, apply it to the service call
+      userLookup.map {
+        case Some(user) => serviceCall(user)
+        case None => throw Forbidden("User must be authenticated to access this service call")
+      }
```

Comment:
I think this should be `Unauthorized` instead of `Forbidden`. I think `401 Unauthorized` is meant to alert that credentials are required but missing or provided but invalid.
This also applies to Java documentation.